runtime/gc_structs/ValueTypesIterator.hpp

```diff
@@ -0,0 +1,78 @@
+
+/*******************************************************************************
+ * Copyright (c) 1991, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Structs
+ */
+
+#if !defined(VALUETYPESITERATOR_HPP_)
+#define VALUETYPESITERATOR_HPP_
+
+#include "j9.h"
+#include "j9cfg.h"
+#include "modron.h"
+
+/**
+ * State constants representing the current stage of the iteration process
+ * @anchor ClassIteratorState
+ */
+enum {
+	valuetypesiterator_state_start = 0,
+	valuetypesiterator_state_default_value,
+	valuetypesiterator_state_end
+};
+
+/**
+ * Iterate over all MethodType slots in a class which contain an object reference.
+ * @ingroup GC_Structs
+ */
+class GC_ValueTypesIterator
+{
+private:
+	int _state;
```

Comment:
if you typedef the enum above to `ValueTypesIteratorState` then you can use it here instead of `int`

Reply:
I agree with @tajila - set and use enum type instead of `int`. Also please move enum to private section of the class and add comment to declaration:
```
class GC_ValueTypesIterator
	{
	private:
                typedef enum {
		       valuetypesiterator_state_start = 0,
		       valuetypesiterator_state_default_value,
		        valuetypesiterator_state_end
	        } ValueTypesState;
                ValueTypesState _state;        /** current Iterator state  */
```

Reply:
In this way, it seems the _state += 1 does not work since the operands are of type ValueTypesState and int. Should I just do _state = valuetypesiterator_state_default_value and _state = valuetypesiterator_state_end, Or do I overload the "+=" operator, or there's a better way to do this?